web/ui/react-app/src/pages/Rules.tsx

```diff
@@ -1,15 +1,108 @@
 import React, { FC } from 'react';
 import { RouteComponentProps } from '@reach/router';
 import PathPrefixProps from '../PathPrefixProps';
-import { Alert } from 'reactstrap';
-
-const Rules: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => (
-  <>
-    <h2>Rules</h2>
-    <Alert color="warning">
-      This page is still under construction. Please try it in the <a href={`${pathPrefix}/rules`}>Classic UI</a>.
-    </Alert>
-  </>
-);
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+import { Alert, Table } from 'reactstrap';
+import { useFetch } from '../utils/useFetch';
+import { Link } from '@reach/router';
+
+interface Rule {
```

Comment:
There are already rule-API-related types defined here: https://github.com/prometheus/prometheus/blob/ae93bae88fed884f59adef604ea059deff81650b/web/ui/react-app/src/pages/alerts/AlertContents.tsx#L9-L45

...but currently they are pretty alerts-specific, as that is used only on the /alerts page, that filters only by alerting rules (https://github.com/prometheus/prometheus/blob/ae93bae88fed884f59adef604ea059deff81650b/web/ui/react-app/src/pages/alerts/Alerts.tsx#L11).

The best thing to do would be to refactor these types into a common location (probably within https://github.com/prometheus/prometheus/tree/master/web/ui/react-app/src/types) where they support both types of rules properly, but are defined only once.

Reply:
Done.